src/components/Alert/Alert.vue

```diff
@@ -1,53 +1,146 @@
+    data-testid="alert"
   >
-    <component
-      :is="icon"
-      size="l"
-      :class="[{ 'text-gray-500': success },
-               { 'text-turquoise-500': info },
-               { 'text-warning': warning },
-               { 'text-coral-700': error }
-      ]"
-    />
-    <div class="px-4">
-      <slot :link-color="linkColor" />
+    <div
+      v-if="hasHeading"
+      class="flex justify-between"
+    >
+      <div class="flex items-center">
+        <component
+          :is="variantDetails.icon"
+          v-if="showIcon"
+          class="h-5 w-5 flex-shrink-0 mr-2"
+          :class="variantDetails.color"
+          data-testid="alertIcon"
+        />
+        <div class="type-base-700">
+          <slot name="heading" />
+        </div>
+      </div>
+      <CloseButton
+        v-if="showCloseButton"
+        @close="closeAlert"
+      />
+      <LearnMoreLink
+        v-if="learnMoreLink && !hasContent"
+        :learn-more-link="learnMoreLink"
+        :link-display-text="linkDisplayText"
+        :show-close-button="showCloseButton"
+        @close="closeAlert"
+      />
+    </div>
+    <div
+      v-if="hasContent"
+      class="flex items-center justify-between"
+    >
+      <div class="flex items-center">
+        <component
+          :is="variantDetails.icon"
+          v-if="showIcon && !hasHeading"
+          class="h-5 w-5 flex-shrink-0 mr-2"
+          :class="variantDetails.color"
+          data-testid="alertIcon"
+        />
+        <div :class="['type-xs-400', {'mt-1': hasHeading && hasContent}]">
+          <slot /> <!-- text/any content goes in the default slot -->
+        </div>
+      </div>
+      <CloseButton
+        v-if="showCloseButton && !hasHeading"
+        @close="closeAlert"
+      />
+    </div>
+    <div class="w-full flex justify-end">
+      <LearnMoreLink
+        v-if="learnMoreLink && hasContent"
+        :learn-more-link="learnMoreLink"
+        :link-display-text="linkDisplayText"
+      />
     </div>
   </div>
 </template>
 
 <script>
+import { Info, Checkmark, AlertCircle, XmarkLarge, ArrowLongRight } from '@/components/Icons';
+import LobLink from '../Link/Link';
 import CircleInfo from '../Icons/CircleInfo.vue';
 import CircleCheck from '../Icons/CircleCheck.vue';
 import CircleExclamation from '../Icons/CircleExclamation.vue';
 
+const LearnMoreLink = {
+  template: `<LobLink
+          :to="learnMoreLink"
+          :underline="false"
+          target="_blank"
+          role="link"
+          class="ml-4 !type-xs-400 text-gray-500"
+        >
+          {{ linkDisplayText }}
+          <ArrowLongRight class="inline ml-0.5 mb-0.5" />
+        </LobLink>`,
+  components: { LobLink, ArrowLongRight },
+  props: {
+    learnMoreLink: String,
+    linkDisplayText: String
+  }
+};
+
+const CloseButton = {
+  template: `
+  <button>
+    <XmarkLarge
+      size="s"
+      data-testid="closeButton"
+      @click="closeAlert"
+    />
+  </button>`,
```

Comment:
Since this button is just an icon without any text it needs an `aria-label` so the screenreader will be able to describe it.